lib/src/react_client/event_helpers.dart

```diff
@@ -564,60 +736,74 @@ SyntheticWheelEvent createSyntheticWheelEvent({
-    bubbles ?? baseEvent?.bubbles ?? false,
-    cancelable ?? baseEvent?.cancelable ?? true,
-    currentTarget ?? baseEvent?.currentTarget,
-    defaultPrevented ?? baseEvent?.defaultPrevented ?? false,
-    preventDefault ?? baseEvent?.preventDefault ?? () {},
-    stopPropagation ?? baseEvent?.stopPropagation ?? () {},
-    eventPhase ?? baseEvent?.eventPhase,
-    isTrusted ?? baseEvent?.isTrusted ?? false,
-    nativeEvent ?? baseEvent?.nativeEvent,
-    target ?? baseEvent?.target,
-    timeStamp ?? baseEvent?.timeStamp ?? 0,
-    type ?? baseEvent?.type ?? 'empty event',
-    deltaX ?? baseEvent?.deltaX,
-    deltaMode ?? baseEvent?.deltaMode,
-    deltaY ?? baseEvent?.deltaY,
-    deltaZ ?? baseEvent?.deltaZ,
-  );
+  return jsifyAndAllowInterop({
+    ..._wrapBaseEventPropertiesInMap(
+      baseEvent: baseEvent,
+      bubbles: bubbles,
+      cancelable: cancelable,
+      currentTarget: currentTarget,
+      defaultPrevented: defaultPrevented,
+      preventDefault: preventDefault,
+      stopPropagation: stopPropagation,
+      eventPhase: eventPhase,
+      isTrusted: isTrusted,
+      nativeEvent: nativeEvent,
+      target: target,
+      timeStamp: timeStamp,
+      type: type,
+    ),
+    'deltaX': deltaX ?? baseEvent?.deltaX,
+    'deltaMode': deltaMode ?? baseEvent?.deltaMode,
+    'deltaY': deltaY ?? baseEvent?.deltaY,
+    'deltaZ': deltaZ ?? baseEvent?.deltaZ,
+  }) as SyntheticWheelEvent;
 }
 
 extension SyntheticEventTypeHelpers on SyntheticEvent {
-  /// Returns whether this is a [SyntheticClipboardEvent].
-  bool get isClipboardEvent => this is SyntheticClipboardEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticClipboardEvent].
+  bool get isClipboardEvent => hasProperty(this, 'clipboardData');
+
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticKeyboardEvent].
+  bool get isKeyboardEvent => hasProperty(this, 'key');
 
-  /// Returns whether this is a [SyntheticKeyboardEvent].
-  bool get isKeyboardEvent => this is SyntheticKeyboardEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticCompositionEvent].
+  bool get isCompositionEvent => hasProperty(this, 'data');
 
-  /// Returns whether this is a [SyntheticCompositionEvent].
-  bool get isCompositionEvent => this is SyntheticCompositionEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticFocusEvent].
+  bool get isFocusEvent => hasProperty(this, 'relatedTarget') && !hasProperty(this, 'button');
 
-  /// Returns whether this is a [SyntheticFocusEvent].
-  bool get isFocusEvent => this is SyntheticFocusEvent;
+  /// Uses Duck Typing to detect if the event instance is a [SyntheticFormEvent].
+  ///
+  /// __NOTE:__ This getter is here for completeness, but because the interface for form events
+  /// is the same as that of a [SyntheticEvent] (the base for all other synthetic event types),
+  /// via Duck Typing every [SyntheticEvent] is considered a [SyntheticFormEvent].
```

Comment:
Hmm if this is the case, I feel like we should just omit it so that consumers don't use it without reading the full doc comment and assume it does something other than always return true.

Or, another thought, could we use the set of event types to determine what type of event it is?
From https://reactjs.org/docs/events.html#form-events:
```
onChange onInput onInvalid onReset onSubmit 
```


But, taking a look on SourceGraph, it's only used in one test, and could be replaced with a not-null-check, so I'd lean toward removing it.

Reply:
I can remove it! 

With removing it, we'd want to go back and deprecate it in the stable line, right?

Reply:
Removed in 067e282!

Reply:
> With removing it, we'd want to go back and deprecate it in the stable line, right?

Ooh, nice catch, I didn't think about that. Yup!